addons/stock_picking_wave/models/stock_picking_wave.py

```diff
@@ -0,0 +1,255 @@
+
+    # -------------------------------------------------------------------------
+    # Action methods
+    # -------------------------------------------------------------------------
+    def action_cancel(self):
+        self.ensure_one()
+        self.state = 'cancel'
+        return True
+
+    def action_done(self):
+        self._check_company()
+        if not self.env.context.get('button_validate_wave_ids'):
+            self = self.with_context(button_validate_wave_ids=self.ids)
+        res = self._pre_action_done_hook()
+        if res is not True:
+            return res
+        new_move_vals = []
+        for move in self.move_line_ids.move_id:
+            if move.move_line_ids <= self.move_line_ids:
+                continue
+            qty = sum([line.qty_done for line in move.move_line_ids & self.move_line_ids])
+            qty_split = move.product_uom._compute_quantity(move.product_uom_qty - qty, move.product_id.uom_id, rounding_method='HALF-UP')
+            new_move = move._split(qty_split, waves=self)
+            new_move_vals += new_move
+        if new_move_vals:
+            new_moves = self.env['stock.move'].create(new_move_vals)
+            new_moves._action_confirm(merge=False)
+        self.move_line_ids.move_id._action_done()
+        return True
+
+    def action_put_in_pack(self):
+        """ Action to put move lines with 'Done' quantities into a new pack
+        This method follows same logic to stock.picking.
+        """
+        self.ensure_one()
+        if self.state not in ('done', 'cancel'):
+            move_line_ids = self.move_line_ids.filtered(
+                lambda ml:
+                float_compare(ml.qty_done, 0.0, precision_rounding=ml.product_uom_id.rounding) > 0
+                and not ml.result_package_id
+            )
+            if not move_line_ids:
+                move_line_ids = self.move_line_ids.filtered(
+                    lambda ml:
+                    float_compare(ml.product_uom_qty, 0.0, precision_rounding=ml.product_uom_id.rounding) > 0
+                    and float_compare(ml.qty_done, 0.0, precision_rounding=ml.product_uom_id.rounding) == 0
+                )
+            if move_line_ids:
+                res = self._pre_put_in_pack_hook(move_line_ids)
+                if not res:
+                    res = self.move_line_ids.picking_id[0]._put_in_pack(move_line_ids, False)
+                return res
+            raise UserError(_("Please add 'Done' quantities to the wave picking to create a new pack."))
+
+    # -------------------------------------------------------------------------
+    # Miscellaneous
+    # -------------------------------------------------------------------------
+    def _pre_put_in_pack_hook(self, move_line_ids):
+        res = self.move_line_ids.picking_id[0]._check_destinations(move_line_ids)
+        if res and res['res_id']:
+            self.env['stock.package.destination'].browse(res['res_id']).wave_id = self
+        return res
```

Comment:
you could override _check_destination instead an add 'wave_id' in the returned action